functions/_common/explorerHandlers.ts

```diff
@@ -0,0 +1,216 @@
+import ReactDOMServer from "react-dom/server"
+import { Env, extensions } from "./env.js"
+import { extractOptions, ImageOptions } from "./imageOptions.js"
+import {
+    buildExplorerProps,
+    Explorer,
+    ExplorerChoiceParams,
+} from "@ourworldindata/explorer"
+import { renderSvgToPng } from "./grapherRenderer.js"
+import { error, png } from "itty-router"
+import { createZip, File } from "littlezipper"
+import { Bounds, Url } from "@ourworldindata/utils"
+import {
+    getSelectedEntityNamesParam,
+    GrapherState,
+    migrateSelectedEntityNamesParam,
+    SelectionArray,
+} from "@ourworldindata/grapher"
+import {
+    assembleCsv,
+    assembleReadme,
+    ensureDownloadOfDataAllowed,
+} from "./downloadFunctions.js"
+import { assembleMetadata } from "./metadataTools.js"
+
+async function initGrapherForExplorerView(
+    env: Env,
+    options: ImageOptions
+): Promise<{
+    grapherState: GrapherState
+    explorerParams: ExplorerChoiceParams
+}> {
+    const url = env.url
+    const explorerPage = await env.ASSETS.fetch(url, { redirect: "manual" })
+
+    const html = await explorerPage.text()
+    const queryStr = url.searchParams.toString()
+    // The env URL class isn't compatible with the Url class from @ourworldindata/utils
+    const urlObj = Url.fromURL(url.toString())
+    const [windowEntityNames] = [urlObj]
+        .map(migrateSelectedEntityNamesParam)
+        .map(getSelectedEntityNamesParam)
```

Comment:
I can see you just copied this existing code, but could we change this to normal function calls? I find this pattern highly unusual and hard to read.